### PR TITLE
Refactor user fields

### PR DESCRIPTION
**Pull request by Piwakk**
Please merge #173  before

### CODE REVIEW

File: backend/authentication/serializers/questions_short.py [hunk 1]
```diff
@@ -1,12 +1,27 @@
...
     question = serializers.StringRelatedField()
+    question_id = serializers.SerializerMethodField()
+
+    def get_question_id(self, answer: ProfileAnswer):
+        return answer.question.id
 
     class Meta:
         model = ProfileAnswer
-        read_only_fields = ("id", "question", "text")
+        read_only_fields = ("id", "question_id", "question", "text")
```
Comment (abocquet) [inline backend/authentication/serializers/questions_short.py]: Don't we expect to have the `question` id in `question`?

File: frontend/src/components/user/edit_profile/EditUserProfile.tsx [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+import * as Yup from "yup";
+import { SelectUsers } from "../../utils/forms/SelectUsers";
+
+// The items in the select.
+const currentAcademicYearItems = new Map([
+    ["1A", <span className="selectgroup-button">1A</span>],
+    ["2A", <span className="selectgroup-button">2A</span>],
+    ["GAP YEAR", <span className="selectgroup-button">Césure</span>],
+    ["3A", <span className="selectgroup-button">3A</span>],
+    ["GRADUATE", <span className="selectgroup-button">Alumni</span>],
```
Comment (abocquet) [inline frontend/src/components/user/edit_profile/EditUserProfile.tsx]: Maybe the span part is a bit redundant

File: backend/authentication/serializers/questions_short.py (hunk 1 above)
Comment (Piwakk) [reply backend/authentication/serializers/questions_short.py]: Flat is better than nested

Comment (Piwakk) [reply backend/authentication/serializers/questions_short.py]: This is why I did it like this

Comment (Piwakk) [reply backend/authentication/serializers/questions_short.py]: question here is just a string

File: frontend/src/components/user/edit_profile/EditUserProfile.tsx (hunk 2 above)
Comment (Piwakk) [reply frontend/src/components/user/edit_profile/EditUserProfile.tsx]: Yes, it's an issue with the component which use them… Let's keep it this way, I opened #179

**Pull request merged**